Makefile

```diff
@@ -386,13 +386,12 @@ test-e2e-set: ## - Run the blackbox end to end tests without setup.
 ##################################################
 .PHONY: test-cloude2e
 test-cloude2e: prepare-test-context  ## - Run cloude2e tests with full setup (slow!)
-	@make -C ${CLOUD_TESTING_BASE} cloud-deploy
-	$(eval FLEET_SERVER_URL := $(shell make -C ${CLOUD_TESTING_BASE} cloud-get-fleet-url))
-	-@set -o pipefail; $(MAKE) test-cloude2e-set | tee build/test-cloude2e.out
-	@make -C ${CLOUD_TESTING_BASE} cloud-clean
+	@# Triggered using a shell script to ensure deployment is cleaned up even if errors (using trap).
+	@# it would also ensure to exit with failure if any error happens
+	@CLOUD_TESTING_BASE=$(CLOUD_TESTING_BASE) dev-tools/cloud/launch_cloud_e2e.sh
 
 .PHONY: test-cloude2e-set
 test-cloude2e-set: ## Run cloude2e test
-	$(eval FLEET_SERVER_URL := $(shell make -C ${CLOUD_TESTING_BASE} cloud-get-fleet-url))
+	$(eval FLEET_SERVER_URL := $(shell make --no-print-directory -C ${CLOUD_TESTING_BASE} cloud-get-fleet-url))
```

Comment:
FLEET_SERVER_URL received as value the whole output of the `make` execution, even the logs about changing directories.

```
FLEET_SERVER_URL=make[1]: Entering directory '/opt/buildkite-agent/builds/bk-agent-prod-gcp-1706643260761682044/elastic/fleet-server/dev-tools/cloud' "" make[1]: Leaving directory '/opt/buildkite-agent/builds/bk-agent-prod-gcp-1706643260761682044/elastic/fleet-server/dev-tools/cloud' go test -v -tags=cloude2e -count=1 -race -p 1 ./testing/cloude2e
bash: Entering: command not found
```

https://buildkite.com/elastic/fleet-server/builds/3039#018d5bce-70ad-42b5-91d9-bfa830da1ad3/90-381

Reply:
Would it be easier to control the output if we use a script instead of a make target for this?

Reply:
How come this is an issue in the context of this PR and happening already in main?

Reply:
@alexsapran In main it's also happening, for instance this build is from one of the latest PRs merged in main

https://buildkite.com/elastic/fleet-server/builds/3068#018d6100-e419-44b9-9ece-bbc987226c94/60-350

```
FLEET_SERVER_URL=make[1]: Entering directory '/opt/buildkite-agent/builds/bk-agent-prod-gcp-1706730452766139194/elastic/fleet-server/dev-tools/cloud' "" make[1]: Leaving directory '/opt/buildkite-agent/builds/bk-agent-prod-gcp-1706730452766139194/elastic/fleet-server/dev-tools/cloud' go test -v -tags=cloude2e -count=1 -race -p 1 ./testing/cloude2e
bash: Entering: command not found
```
It has the same error. However, with te current code in main that error is not handled and the script finishes without any failure.

Reply:
I tried to update the script but it becomes more difficult to understand. I'd prefer to keep it with the `--print-no-directories` flag. @jsoriano 

Revisiting the `dev-tools/cloud/Makefile` , I've updated the target `cloud-get-fleet-url` to just run one terraform command without the eval to show the URL and remove there the double quotes returned by terraform.
https://github.com/elastic/fleet-server/pull/3243/commits/ca8dba3ac38047b68e73d551bb59e6371d936c42